cli/src/main/java/com/devonfw/tools/ide/commandlet/VersionCommandlet.java

```diff
@@ -0,0 +1,34 @@
+package com.devonfw.tools.ide.commandlet;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.property.FlagProperty;
+import com.devonfw.tools.ide.version.IdeVersion;
+
+/**
+ * Prints the IDEasy version and exits
+ */
+public class VersionCommandlet extends Commandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public VersionCommandlet(IdeContext context) {
+
+    super(context);
+    addKeyword(new FlagProperty(getName(), true, "-v"));
+  }
+
+  @Override
+  public String getName() {
+
+    return "--version";
+  }
```

Comment:
Why did we create a new class for Version, and not use it in ContextCommandlet anymore?

Reply:
IMHO the question is more why `--version` was part of the `ContextCommandlet` before since this is triggering to print the version of `IDEasy` that has nothing to do with initializing the context what is the purpose of `ContextCommandlet`.
So initially this was just a pragmatic solution to put it into `ContextCommandlet`.

Now that we implemented auto-completion this got into the way:
* We have auto-completion for bash that needs to include all args and options including the ones from `ContextCommandlet` (e.g. `--force|-f`, `--offline|-o`, etc.)
* We have auto-completion shell via jline3 added by this PR. The shell will be started with `ide [-f] [-d|-t|-q] [-o] shell` and we will create `IdeContext` based on these initial flags given to the `ShellCommandlet`. Within the completion shell you can do auto-completion but since `IdeContext` is already there and it does not make so much sense to change it on the fly (e.g. lower or raise log-level) these options are not suggested/completed here.
* But printing the IDEasy version should always be possible both with `ide --version` and via `ide shell` followed by `ide> --version` it was necessary to extract `VersionCommandlet`.

Thanks for asking questions. I hope the answer was helpful.
@aBega2000 please resolve this conversation after reading.